Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -373,89 +373,107 @@ static void bufferWasConsumedCallback(CMNotificationCenterRef, const void* liste
     resetParserState();
 }
 
-void SourceBufferPrivateAVFObjC::didParseInitializationData(InitializationSegment&& segment)
+void SourceBufferPrivateAVFObjC::setTrackChangeCallbacks(const Vector<RefPtr<TrackPrivateBase>>& tracks, bool initialized)
 {
-    ALWAYS_LOG(LOGIDENTIFIER);
+    for (auto& track : tracks) {
+        if (auto videoTrack = dynamicDowncast<VideoTrackPrivate>(track)) {
```

Comment:
We could use switch(track.type()). Do we need to handle text variant?

Reply:
text tracks can exists, but they can't have any samples though.

Reply:
I will still need to call downcast<> to call the methods, those aren't virtual so it doesn't help. rather than using an array of the base class, I could use an array of variant, but I think it would make it all more complicated than it need to be.